src/pocketmine/entity/Effect.php

```diff
@@ -441,6 +441,9 @@ public function add(Entity $entity, Effect $oldEffect = null){
 					$entity->setAbsorption($new);
 				}
 				break;
+			case Effect::FIRE_RESISTANCE:
+				$entity->extinguish(2);
```

Comment:
This behaviour is obscure. Modifying the fire ticks directly would be better (I'll add methods for this before the PR is merged).